frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,517 @@
+          letterSpacing="0.1em"
+          pos="relative"
+          w="full"
+          pb="30px"
+          as="h2"
+          fontFamily="Roboto Mono"
+          lineHeight="1.15em"
+          _before={
+            sectionTitle
+              ? {
+                  content: "''",
+                  display: "block",
+                  h: "2px",
+                  w: "60px",
+                  pos: "absolute",
+                  top: "7px",
+                  left: "-68px",
+                  backgroundColor: "rgb(129, 82, 188)",
+                }
+              : undefined
+          }>
+          <InlineTextarea name="sectionTitle" />
+        </Box>
+
+        <Flex
+          flexDir={{
+            base: "column",
+            md: "row",
+          }}>
+          <Flex
+            flex={{
+              base: "0 0 full",
+              md: "0 0 300px",
+              xl: "0 0 400px",
+            }}
+            mr={{
+              base: "0",
+              md: "20",
+              lg: "36",
+            }}
+            flexDir="column">
+            <Box
+              as="h2"
+              fontSize={{ base: "3xl", sm: "4xl", md: "5xl" }}
+              fontFamily="Europa"
+              fontWeight="bold"
+              lineHeight="1.1em">
+              <InlineTextarea name="title" />
+            </Box>
+            <Text fontSize="sm" pt="5" color="description">
+              <InlineTextarea name="subtitle" />
+            </Text>
+            <Link href="mailto:DA INSERIRE" passHref>
+              <Box
+                as="a"
+                fontSize="sm"
+                pt="14"
+                lineHeight="1.8em"
+                textDecoration="none">
+                <InlineTextarea name="email" />
+              </Box>
+            </Link>
+          </Flex>
+          <ContactsForm />
+        </Flex>
+
+        <Flex justifyContent="flex-end" w="full" mb="32">
+          <Social
+            isBubbleActive={areBubblesActive}
+            link="https://twitter.com/inkofpixel"
+            color="rgba(29, 161, 242, 0.7)"
+            onHoverColor="rgba(29, 161, 242, 1)">
+            <TwitterIcon color="white" />
+          </Social>
+          <Social
+            isBubbleActive={areBubblesActive}
+            link="https://facebook.com/inkofpixel"
+            color="rgba(59, 89, 152, 0.7)"
+            onHoverColor="rgba(59, 89, 152, 1)">
+            <FacebookIcon color="white" />
+          </Social>
+          <Social
+            isBubbleActive={areBubblesActive}
+            link="https://github.com/inkofpixel"
+            color="rgba(24, 23, 23, 0.7)"
+            onHoverColor="rgba(24, 23, 23, 1)">
+            <GithubIcon color="white" />
+          </Social>
+        </Flex>
+      </Flex>
+    </Box>
+  );
+}
+
+const blankForm = {
+  name: "",
+  email: "",
+  message: "",
+  isSubmitted: false,
+};
+
+function reducer(state: any, action: any) {
+  switch (action.type) {
+    case "setName":
```

Comment:
1. It doesn't really make sense to have an action for every field (e.g. `setName`, `setEmail`, etc), since all the actions are doing pretty much the same thing, update a field.. So let's have an action `update-field` that receives the field name and the new value, and updates state accordingly
2. You are not handling what happens if a submission error occurs 
3. `action.type` should be an enum:
    ```ts
    enum FormActionType {
      UpdateField = "update-field",
      Success = "success",
      Failed = "failed"
    }
    ```